Create bookings management page

denjj opened the pull request:
### What this PR does (required):
- Creates components for viewing a sitter's bookings and their calendar
- Also includes accept/decline booking and grouping between past or future bookings
- Adds frontend api calls for obtaining bookings and responding with approve/decline
- Designates interface for Request data, to be used with request api
- Closes #19 and #31

### Screenshots / Videos (front-end only):
![msedge_reZKXhtIP0](https://user-images.githubusercontent.com/25715300/153681829-7b0f3978-e134-4370-9d87-8afded114f35.gif)



### Any information needed to test this feature (required):
- Sign up two new users
- Create booking requests from user A -> B through an api tool while authenticated as user A
	POST /requests/create with sitterId, start (Date), end (Date) in the body
- Log in as user B, click on the My Sitters link in the navbar and new bookings page will be shown
- Click on the cog to open the menu to approve/deny a booking request

### Any issues with the current functionality (optional):
- The calendar is currently a bit smaller than what is shown from the mockup. I used the mui static date picker component, but the documentation was lacking information on how to resize the calendar view. I tried to brute force a solution through various css selectors and forced theming, but could not find a way to make the calendar bigger.

ezeilo-su reviewed (commented):
Hey @denjj, good work with this one.

denjj commented:
I rebased and updated portions of this branch to remain compatible with changes in request controller routes from #55 and request model from [0e20fb8](https://github.com/hatchways/team-watermint/pull/53/commits/0e20fb867e1896a355fbf1c72658ec7841f4fd1f)

ezeilo-su commented:
Hey @denjj, kindly resolve merge conflicts.

rajivtitus commented on client/src/pages/Bookings/Booking/Booking.tsx:
```diff
@@ -0,0 +1,81 @@
+interface Props {
+  start: string;
+  end: string;
+  userId: User;
```
```suggestion
  user: User;
```
This seems like a user object and not just the id - ditto throughout :)

rajivtitus commented on client/src/interface/RequestApiData.ts:
```diff
@@ -0,0 +1,20 @@
+import { RequestStatus } from '../types/RequestStatus';
+export interface Request {
+  _id: string;
+  userId: User;
```
```suggestion
  user: User;
```

rajivtitus commented on client/src/pages/Bookings/Booking/Booking.tsx:
```diff
@@ -0,0 +1,81 @@
+  border = false,
+  handleRequest: handleRequest,
+}: Props): JSX.Element {
+  // function renderRequestStatus(status: RequestStatus): string {
```
Please uncomment or remove commented code

rajivtitus commented on client/src/pages/Bookings/Calendar/Calendar.tsx:
```diff
@@ -0,0 +1,34 @@
+  const highlightedDays = requests.map((x) => new Date(x.start).toDateString());
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
```
We can move the `LocalizationProvider` to our App.tsx so that it can be used throughout our app

rajivtitus commented on client/src/pages/Bookings/Calendar/Calendar.tsx:
```diff
@@ -0,0 +1,34 @@
+
+export default function Calendar({ requests }: Props): JSX.Element {
+  const initialDate = new Date();
+  const highlightedDays = requests.map((x) => new Date(x.start).toDateString());
```
```suggestion
  const highlightedDays = requests.map((request) => new Date(request.start).toDateString());
```
Nit! 😄

rajivtitus commented on client/src/pages/Bookings/Booking/Booking.tsx:
```diff
@@ -0,0 +1,81 @@
+        spacing={3}
+      >
+        <Grid item xs={9}>
+          <Typography sx={{ fontWeight: 'bold', fontSize: 16 }}>
```
As opposed to using the `sx` prop to style `Typography` components inline, we can set 'font - weights and sizes' for variants in MUI's theme and use those variants throughout the app :)

rajivtitus commented on client/src/pages/Bookings/Booking/EditBooking.tsx:
```diff
@@ -0,0 +1,67 @@
+        }}
+      >
+        <Button
+          onClick={() => {
```
An inline function in render allocates a new function instance each time render is called. Some suggest that this will cause performance issues because the garbage collector will be be invoked more often. As a result, many React developers encourage you to never use inline functions in render :)

Ditto below!

rajivtitus commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,146 @@
+import useStyles from './useStyles';
+import { RequestStatus } from '../../types/RequestStatus';
+
+export default function Sitters(): JSX.Element {
```
```suggestion
export default function Bookings(): JSX.Element {
```
Wouldn't Booking/Request be a more suitable name for this component

rajivtitus commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,146 @@
+  }, []);
+
+  function handleRequest(requestId: string, status: RequestStatus) {
+    updateRequest(requestId, status).then(() => {
```
In case an update has failed, we can update the Snackbar with the appropriate error message

rajivtitus commented on client/src/pages/Bookings/Bookings.tsx:
```diff
@@ -0,0 +1,146 @@
+        <Grid container direction="column" spacing={3}>
+          <Grid item>
+            <Paper elevation={3} sx={{ paddingX: 4, paddingY: 4, minWidth: 450 }}>
+              <Typography sx={{ fontWeight: 'bold', fontSize: 13, textTransform: 'uppercase' }}>
```
Same as comment above about the usage of the `sx` prop

rajivtitus reviewed (approved):
Excellent work with this! Great use of React and MUI 💯 
Just some minor comments to address before merging 👍🏽
